### PR TITLE
1. Move enable RMI to be a function for further use. 2. Implement core write/read functions for further use.

### PR DESCRIPTION
**Pull request by blueue**
1. Move enable RMI to be a function for further use.
2. Implement core write/read functions for further use.

### CODE REVIEW

File: plugins/synaptics-rmi/fu-synaptics-rmi-firmware.c [hunk 1]
```diff
@@ -379,6 +384,13 @@ fu_synaptics_rmi_firmware_parse (FuFirmware *firmware,
 	case 6:
 		if (!fu_synaptics_rmi_firmware_parse_v0x (firmware, fw, error))
 			return FALSE;
+		self->hasBlv5Signature = ((self->io & 0x10) >> 1);
+		if(self->hasBlv5Signature){
```
Comment (hughsie) [inline plugins/synaptics-rmi/fu-synaptics-rmi-firmware.c:388]: space after `if` space before `{`

File: plugins/synaptics-rmi/fu-synaptics-rmi-firmware.c [hunk 2]
```diff
@@ -379,6 +384,13 @@ fu_synaptics_rmi_firmware_parse (FuFirmware *firmware,
...
+			for(int i=0 ; i<RMI_IMG_SIGNATURE_SIZE_SIZE ; i++){
```
Comment (hughsie) [inline plugins/synaptics-rmi/fu-synaptics-rmi-firmware.c:389]: spaces around `=` and `<` and also guint please

File: plugins/synaptics-rmi/fu-synaptics-rmi-ps2-device.c [hunk 3]
```diff
@@ -97,6 +97,140 @@ WriteByte (FuSynapticsRmiPs2Device *self, guint8 buf, guint timeout, GError **er
...
+static gboolean
+WriteRMIRegister (FuSynapticsRmiPs2Device *self, 
+				 guint8 addr, 
+				 const guint8 *buf, 
+				 guint8 len, 
+				 guint timeout, 
+				 GError **error)
+{
+
+	if (!self->inRMIBackdoor && !fu_synaptics_rmi_ps2_device_enable_RMI_backdoor(self, error)) {
```
Comment (hughsie) [inline plugins/synaptics-rmi/fu-synaptics-rmi-ps2-device.c:109]: always space between function name and bracket pls

File: plugins/synaptics-rmi/fu-synaptics-rmi-ps2-device.c [hunk 4]
```diff
@@ -97,6 +97,140 @@ WriteByte (FuSynapticsRmiPs2Device *self, guint8 buf, guint timeout, GError **er
...
+
+static gboolean
+ReadRMIRegister (FuSynapticsRmiPs2Device *self, 
+				guint8 addr, 
+				guint8 * buf,
+				GError **error)
+{
+	if (!buf) {
+		return FALSE;
+	}
```
Comment (hughsie) [inline plugins/synaptics-rmi/fu-synaptics-rmi-ps2-device.c:156]: we normally do `== NULL` for pointers, not `!`

**Comment by hughsie** [comment]
Lots of little fixes with spacing please, and then please push directly to the main origin branch -- I didn't know you were not already a committer, apologies. You should have an invite via email.

File: plugins/synaptics-rmi/fu-synaptics-rmi-ps2-device.c (hunk 4 above)
Comment (superm1) [reply plugins/synaptics-rmi/fu-synaptics-rmi-ps2-device.c:156]: this seems like it should be `g_return_val_if_fail` actually

File: plugins/synaptics-rmi/fu-synaptics-rmi-ps2-device.c [hunk 5]
```diff
@@ -97,6 +97,140 @@ WriteByte (FuSynapticsRmiPs2Device *self, guint8 buf, guint timeout, GError **er
...
+
+	g_debug ("ReadRMIRegister: register address = 0x%x", addr);
+	if (!self->inRMIBackdoor && !fu_synaptics_rmi_ps2_device_enable_RMI_backdoor(self, error)) {
+		g_prefix_error (error, "failed to enable RMI backdoor: ");
+		return FALSE;
+	}
+
+	g_autoptr(GError) error_local = NULL;
```
Comment (superm1) [inline plugins/synaptics-rmi/fu-synaptics-rmi-ps2-device.c:164]: this needs to be declared at start of function

File: plugins/synaptics-rmi/fu-synaptics-rmi-ps2-device.c [hunk 6]
```diff
@@ -97,6 +97,140 @@ WriteByte (FuSynapticsRmiPs2Device *self, guint8 buf, guint timeout, GError **er
...
+	if (!(WriteByte(self, edpAuxSetScaling2To1, 0, &error_local) && WriteByte(self, edpAuxSetSampleRate, 0, &error_local)
+		&& WriteByte(self, addr, 0, &error_local) && WriteByte(self, edpAuxStatusRequest, 0, &error_local)))
+	{
+		g_warning ("failed to write command in Read RMI Register: %s", error_local->message);
+		return FALSE;
```
[the remark this answers is not in the document]
Comment (hughsie) [reply plugins/synaptics-rmi/fu-synaptics-rmi-ps2-device.c:169]: I've fixed this up for them; they're learning :)

**Pull request closed without merging**